toolkit/cli/smart-contracts-commands/src/register.rs

```diff
@@ -61,3 +64,36 @@ impl RegisterCmd {
 		Ok(())
 	}
 }
+
+#[derive(Clone, Debug, clap::Parser)]
+pub struct DeregisterCmd {
+	#[clap(flatten)]
+	common_arguments: crate::CommonArguments,
+	#[arg(long)]
+	genesis_utxo: UtxoId,
+	#[arg(long)]
+	payment_key_file: String,
+	#[arg(long)]
+	spo_public_key: MainchainPublicKey,
+}
+
+impl DeregisterCmd {
+	pub async fn execute(self) -> crate::CmdResult<()> {
+		let payment_signing_key = read_private_key_from_file(&self.payment_key_file)?;
+		let client = HttpClient::builder().build(self.common_arguments.ogmios_url)?;
+		let own_pkh =
+			crate::payment_signing_key_to_mainchain_address_hash(payment_signing_key.clone())?;
```

Comment:
1. I think payment signing key is enough to pass to run_register.
2. Now I see that the name `payment_signing_key_to_mainchain_address_hash` amplifies confusion of wrong type name. McAddressHash is in fact McKeyHash or McPublicKeyHash. It has nothing to do with address.
This is one of things that we should clean up.